addons/mail/static/tests/clock_model_tests.js

```diff
@@ -0,0 +1,71 @@
+/** @odoo-module **/
+
+import { insertAndReplace, link } from '@mail/model/model_field_command';
+import { afterEach, beforeEach, start } from '@mail/utils/test_utils';
+
+QUnit.module('mail', {}, function () {
+QUnit.module('clock_model_tests', {
+    async beforeEach() {
+        await beforeEach(this);
+
+        this.start = async params => {
+            const res = await start({ ...params, data: this.data });
+            const { apps, env, widget } = res;
+            this.apps = apps;
+            this.env = env;
+            this.widget = widget;
+            return res;
+        };
+    },
+    afterEach() {
+        afterEach(this);
+    },
+});
+
+QUnit.test('Clock has no interval to begin with.', async function (assert) {
+    assert.expect(1);
+
+    await this.start();
+    const clock = this.messaging.models['Clock'].insert({ frequency: 60 * 1000 });
+
+    assert.strictEqual(
+        clock.intervalId,
+        undefined,
+        "the clock must have no interval to begin with."
+    );
+});
+
+QUnit.test('Adding a clock watcher to a clock creates an interval.', async function (assert) {
+    assert.expect(1);
+
+    await this.start();
+    const clock = this.messaging.models['Clock'].insert({ frequency: 60 * 1000 });
+
+    this.messaging.models['QUnitTest'].insert({
+        clockWatcher: insertAndReplace({ clock: link(clock) }),
+    });
+    assert.notEqual(
+        clock.intervalId,
+        undefined,
+        "the clock must have an interval after a watcher is added."
+    );
+});
+
+QUnit.test('Removing all watchers of a clock clear the interval.', async function (assert) {
+    assert.expect(1);
+
+    await this.start();
+    const clock = this.messaging.models['Clock'].insert({ frequency: 60 * 1000 });
+    const watcher = this.messaging.models['QUnitTest'].insert({
+        clockWatcher: insertAndReplace({ clock: link(clock) }),
+    });
+
+    watcher.delete();
+    assert.strictEqual(
+        clock.intervalId,
+        undefined,
+        "the clock must no longer have an interval after all its watchers have been deleted."
+    );
+});
```

Comment:
Name of test is broad, and scenario is quite specific: here it's just tested with single watched and it's asserting no interval when the watched is deleted.

It assumes single watched deletion => single watcher removed => no more watchers so no interval.